sympy/algebras/tests/test_quaternion.py

```diff
@@ -57,6 +57,11 @@ def test_quaternion_complex_real_addition():
     assert q1 - q1 == q0
 
 
+def test_quaternion_evalf():
+    assert Quaternion(1, 2, 3, 4).evalf() == Quaternion(Float(1), Float(2), Float(3), Float(4))
+    assert Quaternion(1 + 2*I, 0, 0, 3+4*I).evalf() == Quaternion(Float(1) + Float(2)*I, 0, 0, Float(3) + Float(4)*I)
```

Comment:
It would be better to test with something like `sqrt(2)` that isn't already an exact rational number.

Reply:
@oscarbenjamin i'm kinda struck up there because (sqrt(2)).evalf() returns 1.41421356237310 but Quaternion(sqrt(2), 0, 0, 0) returns 1.4142135623731 + 0*i + 0*j + 0*k , could you tell me what mistake am i doing here?

Reply:
I'm not sure but you could add some code to print out the arguments that are being passed to the internal evalf call. You can use `srepr` to see precise value of Floats in an expression.

Reply:
@oscarbenjamin the thing is found is while (sqrt(2)).evalf() is returning 1.41421356237310 inside the _eval_evalf() method but Quaternion((sqrt(2)).evalf()) is returning 1.4142135623731 + 0i + 0j + 0*k and the 0 is omitted after passing through Quaternion while srepr for both (sqrt(2)).evalf() and Quaternion((sqrt(2)).evalf()) is returning Float('1.4142135623730951', precision=53) there is something with returning the Quaternion which i am unable to figure out?

Reply:
If `srepr` shows that the numbers are the same then they should be the same. If they look different then I guess that's just to do with the way that the expressions are printed.

Reply:
@oscarbenjamin sorry , i misunderstood when the way they were printed is different , the test cases are working now could you review the code?
Thanks!